lib/OpenQA/WebAPI/Controller/Step.pm

```diff
@@ -301,16 +300,11 @@ sub _basic_needle_info {
 
     $file_name //= "$name.json";
     $file_name = locate_needle($file_name, $needles_dir) if !-f $file_name;
-    return undef unless defined $file_name;
+    return wantarray ? (undef, 'File not found') : undef unless defined $file_name;
```

Comment:
Can't we just return the array in any case. I don't like the context-sensitive return type.

Reply:
I copied that from @perlpunk :-D The error message is only being used in one case, and ignored in the other ones.

Reply:
I agree with @Martchus here, I don't use `wantarray` that often. In this case the function is internal and only used 3 times.
In the cases where the additional info is not needed, do:
`my ($needle_info) = ...`